torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2581,7 +2586,11 @@ def create_symbol(
             dynamic_dim = DimDynamic.DYNAMIC
 
         if dynamic_dim is DimDynamic.STATIC:
-            return sympy.Integer(val)
+            if isinstance(val, int):
+                return sympy.Integer(val)
+            else:
+                return sympy.Symbol(str(val))
```

Comment:
this is hella suspicious. Why?

Reply:
Creating an Integer requires passing in a plain int, which we don't have.

Reply:
I think it would probably be better not to try to inject singletons into sympy's symbolic expression universe, esp since we don't really support any nontrivial arithmetic on them (except for comparisons against 0/1, but I'm guessing this can be special cased). One reason I'm not keen on treating singletons as sympy Symbols is that symbols don't have correct semantics for singletons:

```
>>> import sympy
>>> from sympy.abc import j0, j1
>>> sympy.Eq(j0, j1)
Eq(j0, j1)
```

This is wrong: the singletons j0 and j1 can NEVER equal each other. But sympy doesn't know that, because to sympy, these are not skolem variables; they could get substituted later.

Reply:
As discussed offline to represent singleton int as a subclass of sympy.AtomExpr instead.